restli-tools/src/test/java/com/linkedin/restli/tools/clientgen/TestRestRequestBuilderGenerator.java

```diff
@@ -121,6 +134,89 @@ public void testGeneration(RestliVersion version, String ABuildersName, String B
     Assert.assertTrue(bBuilderFileContent.contains("Generated from " + RESOURCES_DIR + FS + "idls" + FS + "arrayDuplicateB.restspec.json"));
   }
 
+  /**
+   * Testing case sensitivity of generated files. Typically a Mac/Windows system will have a case insensitive file
+   * system, whereas a Linux system will have a case sensitive file system. For a case insensitive file system,
+   * "~/com/astro" and "~/com/ASTRO" point to the same folder. For a case sensitive file system, "~/com/astro" and "~/com/ASTRO"
+   * will be different folders.
+   *
+   * Example:
+   *   File1: namespace = com.astro.file1
+   *   File2: namespace = com.aSTRo.file2
+   *
+   *   The following files would be generated with the path specified.
+   *   1) Case insensitive (if file1 is generated first):
+   *       com/astro/file1
+   *                /file2
+   *   2) Case insensitive (if file2 is generated first):
+   *       com/aSTRo/file1
+   *                /file2
+   *   3) Case sensitive:
+   *       com/astro/file1
+   *           aSTRo/file2
+   *
+   * @param version RestLi version
+   * @param restspec1 First restli spec to generate
+   * @param restspec2 Second restli spec to generate
+   * @param generateLowercasePath True, generate path lowercase; False, generate path as spec specifies.
+   */
+  @Test(dataProvider = "arrayDuplicateDataProvider2")
+  public void testGenerationPathOrder(RestliVersion version, String restspec1, String restspec2, boolean generateLowercasePath) throws Exception
```

Comment:
Can you also add a simple test to show that your change only affect "package" name, but not the root directory (because some user uses custom root directory instead of the sourceSet directory `<module>/src/main`

Reply:
Added new test.